lib/rpmsg/rpmsg_virtio.c

```diff
@@ -584,6 +584,8 @@ static int rpmsg_virtio_ns_callback(struct rpmsg_endpoint *ept, void *data,
 		metal_mutex_release(&rdev->lock);
 		if (_ept && _ept->ns_unbind_cb)
 			_ept->ns_unbind_cb(_ept);
+		if (rdev->ns_unbind_cb)
```

Comment:
At this point you can not ensure that the endpoint is released, 
Seems more safe to call the call back in rpmsg_destroy_ept

Reply:
This callback just notify the porting layer the endpoint at the remote side was destroyed, not relate to the life cycle of the local endpoint, and it's also symmetric with ns_bind_cb at line 599.

Reply:
Both rpmsg and remoteproc layer need bind/unbind in many case, but bind is only sent to rpmsg_device_ops and unbind is only sent to rpmsg_endpoint in the current code base. PR https://github.com/OpenAMP/open-amp/pull/246 and https://github.com/OpenAMP/open-amp/pull/376 make both layer can receive the bind/unbind notification which make the notification more symmetric and functional.

Reply:
Could you add callback description in https://github.com/OpenAMP/open-amp/blob/main/docs/rpmsg-design.md#rpmsg-user-defined-callbacks documentation?
To help user Please, please add sub chapter for endpoint callbacks and service announcement callbacks

Reply:
Done.